controllers/cache_handler.go

```diff
@@ -511,7 +511,13 @@ func (r *DisruptionReconciler) manageInstanceSelectorCache(instance *chaosv1beta
 
 		// start the cache with a cancelable context and duration, and attach it to the controller as a watch source
 		ch := make(chan error)
-		cacheCtx, cacheCancelFunc := context.WithTimeout(context.Background(), instance.Spec.Duration.Duration()+*r.ExpiredDisruptionGCDelay*2)
+		deletionDelay := time.Minute * 2
```

Comment:
This default value seems risky as a disruption can have no GC expiration delay and last longer than a couple of minutes. In this case, it would clear the cache for a still existing disruption.

Reply:
Agreed. It would then recreate another cache with the disruption duration, and so on until the disruption is deleted. This is not a suitable solution.